pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,465 @@
+        drawer.box_gate(1, [0, 1], "U(θ)")
+        drawer.box_gate(1, 4, "X", color='lightcoral')
+
+        drawer.SWAP(1, (2, 3))
+        drawer.CNOT(2, (0,2), color='forestgreen')
+
+        drawer.ctrl(3, [1,3])
+        drawer.box_gate(3, 2, "H", zorder_base=2)
+
+        drawer.ctrl(4, [1,2])
+
+        drawer.measure(5, 0)
+
+        drawer.fig.suptitle('My Circuit', fontsize='xx-large')
+
+    .. figure:: ../../_static/drawer/example_Solarize_Light2.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+
+    """
+
+    def __init__(self, n_layers, n_wires, figsize=None):
+
+        self.n_layers = n_layers
+        self.n_wires = n_wires
+
+        # half the width of a box
+        # is difference between center and edge
+        self._box_dx = 0.4
+
+        self._circ_rad = 0.3
+        self._ctrl_rad = 0.1
+        self._swap_dx = 0.2
+
+        ## Creating figure and ax
+
+        if figsize is None:
+            figsize = (self.n_layers + 3, self.n_wires + 1)
+
+        self.fig = plt.figure(figsize=figsize)
+        self.ax = self.fig.add_axes(
+            [0, 0, 1, 1],
+            xlim=(-2, self.n_layers + 1),
+            ylim=(-1, self.n_wires),
+            xticks=[],
+            yticks=[],
+        )
+
+        # adding wire lines
+        for wire in range(self.n_wires):
+            line = plt.Line2D((-1, self.n_layers), (wire, wire), zorder=1)
+            self.ax.add_line(line)
+
+    def label(self, labels):
+        """Label each wire.
+
+        Args:
+            labels [Iterable[str]]: Iterable of labels for the wires
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=1)
+            drawer.label(["a", "b"])
+
+        .. figure:: ../../_static/drawer/labels.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        for wire, ii_label in enumerate(labels):
+            self.ax.text(-1.5, wire, ii_label)
+
+    def box_gate(
+        self, layer, wires, text="", extra_width=0, rotate_text=False, zorder_base=0, color=None
+    ):
+        """Draws a box and adds label text to it's center.
+
+        Args:
+            layer (Int)
+            wires (Union[Int, Iterable[Int]])
+            text (str)
+
+        Kwargs:
+            extra_width=0 (float): Extra box width
+            rotate_text=False (Bool): whether to rotate text 90 degrees. Helpful to long labels and
+                multi-wire boxes.
+            zorder_base (Int): shift the object in zorder
+            color=None: mpl compatible color designation
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=2)
+
+            drawer.box_gate(layer=0, wires=0, text="Y")
+            drawer.box_gate(layer=1, wires=(0,1), text="CRy(0.1)", rotate_text=True)
+
+        .. figure:: ../../_static/drawer/box_gates.png
```

Comment:
Suggestion: option for white background on boxgates, with coloured outline (e.g., `transparent=True` kwarg?)

Reply:
Playing around with adding `box_kwargs` and `text_kwargs`, that can accept dictionaries of arbitrary mpl keywords.  But then it starts getting a little clunky maybe?

Reply:
This is great, I was able to accomplish the white background using:
```
box_kwargs = {'facecolor' : 'white', 'edgecolor' : 'black', 'linewidth' : 3}
```
It _is_ a bit clunky, but it'd be more clunky to have everything as separate kwargs. As it is now, you can specify the properties of all boxes / lines / text in the circuits in dictionaries once, then use them throughout, which is nice.

One way to do this in a cleaner way could be to have the drawer object accept and store all these `kwargs`, but still include the option for a manual override of the formatting in each type of gate. So like,

```python
box_kwargs = {'facecolor' : 'white', 'edgecolor' : 'black', 'linewidth' : 3}

drawer = MPLDrawer(n_wires=5,n_layers=5, wire_kwargs=wire_kwargs, box_kwargs=box_kwargs)

# Normal box gate with the parameters in the dictionary above
drawer.box_gate(1, [0, 1], "U(θ)")

# Fancier box gate
drawer.box_gate(2, [0, 1], "U(x)", box_kwargs={'edgecolor' : 'pink'})

```